diff --git a/drivers/hid/hid-input.c b/drivers/hid/hid-input.c
--- a/drivers/hid/hid-input.c
+++ b/drivers/hid/hid-input.c
@@ -98,10 +98,11 @@ static struct hidinput_key_translation apple_fn_keys[] = {
 
 static struct hidinput_key_translation apple_fn_keys[] = {
 	{ KEY_BACKSPACE, KEY_DELETE },
+	{ KEY_ENTER,    KEY_INSERT },
 	{ KEY_F1,       KEY_BRIGHTNESSDOWN, APPLE_FLAG_FKEY },
 	{ KEY_F2,       KEY_BRIGHTNESSUP,   APPLE_FLAG_FKEY },
-	{ KEY_F3,       KEY_FN_F5,          APPLE_FLAG_FKEY }, /* Exposé */
-	{ KEY_F4,       KEY_FN_F4,          APPLE_FLAG_FKEY }, /* Dashboard */
+	{ KEY_F3,       KEY_CYCLEWINDOWS,   APPLE_FLAG_FKEY }, /* Exposé */
+	{ KEY_F4,       KEY_CALC,           APPLE_FLAG_FKEY }, /* Dashboard */
 	{ KEY_F5,       KEY_KBDILLUMDOWN,   APPLE_FLAG_FKEY },
 	{ KEY_F6,       KEY_KBDILLUMUP,     APPLE_FLAG_FKEY },
 	{ KEY_F7,       KEY_PREVIOUSSONG,   APPLE_FLAG_FKEY },
